docs: Add note to check S3 configuration before apply (#316)
elastisys/compliantkubernetes-apps: README.md

3. Initialize your environment and configuration:
   Note that this will *not* overwrite existing values, but it will append to existing files.

>    See the [compliantkubernetes][compliantkubernetes] repository if you are uncertain about in what order you should do things.

Reviewer: Maybe refer to the website instead
```suggestion
   See [compliantkubernetes.io](compliantkubernetes.io) if you are uncertain about in what order you should do things.
```

   ```bash
   ./bin/ck8s init
   ```

4. Edit the configuration files that have been initialized in the configuration path.
   Make sure that the `objectStorage` values are set in `sc-config.yaml`, `wc-config.yaml` and `secrets.yaml` according to your `objectStorage.type` (Set `objectStorage.s3.*` if you are using s3 or `objectStorage.gcs.*` if you are using gcs.)
   If you enable object storage you also need to make sure that the buckets specified in `objecStorage.buckets` exist.
   You can can running the following snippet to ensure that you've configured S3 correctly:

   ```bash

>    for bucket in $(yq r ${CK8S_CONFIG_PATH}/sc-config.yaml 'objectStorage.buckets.*'); do

Reviewer 2: Is this going to use the keys configured in `secrets.yaml`? If not, I see a risk that it will give the user a false sense of safety.

Author: Yes of course that is what ./bin/ck8s uses.

Author: Gah, missed that it uses s3cfg, will fix.

3. Initialize your environment and configuration:
   Note that this will *not* overwrite existing values, but it will append to existing files.
   See [compliantkubernetes.io](compliantkubernetes.io) if you are uncertain about in what order you should do things.

   ```bash
   ./bin/ck8s init
   ```

4. Edit the configuration files that have been initialized in the configuration path.
   Make sure that the `objectStorage` values are set in `sc-config.yaml`, `wc-config.yaml` and `secrets.yaml` according to your `objectStorage.type` (Set `objectStorage.s3.*` if you are using s3 or `objectStorage.gcs.*` if you are using gcs.)
   If you enable object storage you also need to make sure that the buckets specified in `objecStorage.buckets` exist.

>    You can can run the following snippet to ensure that you've configured S3 correctly:

Reviewer 3: ```suggestion
   You can run the following snippet to ensure that you've configured S3 correctly:
```

>    You can can running the following snippet to ensure that you've configured S3 correctly:

Reviewer 3: ```suggestion
   You can run the following snippet to ensure that you've configured S3 correctly:
```

   ```bash
  (
      access_key=$(sops exec-file ${CK8S_CONFIG_PATH}/secrets.yaml 'yq r {} "objectStorage.s3.accessKey"')
      secret_key=$(sops exec-file ${CK8S_CONFIG_PATH}/secrets.yaml 'yq r {} "objectStorage.s3.secretKey"')
      region=$(yq r ${CK8S_CONFIG_PATH}/sc-config.yaml 'objectStorage.s3.region')
      host=$(yq r ${CK8S_CONFIG_PATH}/sc-config.yaml 'objectStorage.s3.regionEndpoint')

      for bucket in $(yq r ${CK8S_CONFIG_PATH}/sc-config.yaml 'objectStorage.buckets.*'); do
          s3cmd --access_key=${access_key} --secret_key=${secret_key} \
              --region=${region} --host=${host} \
              ls s3://${bucket} > /dev/null
          [ ${?} = 0 ] && echo "Bucket ${bucket} exists!"
      done
  )

>    ```

Reviewer 3: It looks like this does not render as expected on github, but I'm not sure why (looks good in VS code). Check if you can fix the markdown formatting to work on github as well.
